src/SixLabors.Fonts/Tables/Cff/Cff1Parser.cs

```diff
@@ -0,0 +1,1256 @@
+            // operators have been derived from FontInfo dict keys but have
+            // been grouped together with the Top DICT operators for
+            // simplicity.The keys from the FontInfo dict are indicated in the
+            // Default, notes  column of Table 9)
+            return this.ReadDICTData(reader, offsets[0].Length);
+        }
+
+        private string[] ReadStringIndex(BigEndianBinaryReader reader)
+        {
+            // 10 String INDEX
+            // All the strings, with the exception of the FontName and
+            // CIDFontName strings which appear in the Name INDEX, used by
+            // different fonts within the FontSet are collected together into an
+            // INDEX structure and are referenced by a 2 - byte unsigned
+            // number called a string identifier or SID.
+
+            // Only unique strings are stored in the table
+            // thereby removing duplication across fonts.
+
+            // Further space saving is obtained by allocating commonly
+            // occurring strings to predefined SIDs.
+            // These strings, known as the standard strings,
+            // describe all the names used in the ISOAdobe and
+            // Expert character sets along with a few other strings
+            // common to Type 1 fonts.
+
+            // A complete list of standard strings is given in Appendix A
+
+            // The client program will contain an array of standard strings with
+            // nStdStrings elements.
+            // Thus, the standard strings take SIDs in the
+            // range 0 to(nStdStrings –1).
+
+            // The first string in the String INDEX
+            // corresponds to the SID whose value is equal to nStdStrings, the
+            // first non - standard string, and so on.
+
+            // When the client needs to
+            // determine the string that corresponds to a particular SID it
+            // performs the following: test if SID is in standard range then
+            // fetch from internal table,
+            // otherwise, fetch string from the String
+            // INDEX using a value of(SID – nStdStrings) as the index.
+            if (!this.TryReadIndexDataOffsets(reader, out CffIndexOffset[]? offsets))
+            {
+                return Array.Empty<string>();
+            }
+
+            string[] stringIndex = new string[offsets.Length];
+
+            // Allow reusing the same buffer for shorter reads.
+            using var buffer = new Buffer<byte>(512);
+            Span<byte> bufferSpan = buffer.GetSpan();
+
+            for (int i = 0; i < offsets.Length; ++i)
+            {
+                int length = offsets[i].Length;
+                if (length < bufferSpan.Length)
+                {
+                    Span<byte> slice = bufferSpan.Slice(0, length);
+                    int actualRead = reader.BaseStream.Read(slice);
+#if DEBUG
```

Comment:
Why is this only thrown when debugging? Is this not considered an error in any case?